prow/tide/tide.go

```diff
@@ -1015,11 +1015,11 @@ func headContexts(log *logrus.Entry, ghc githubClient, pr *PullRequest) ([]Conte
 // struct without making additional API calls. It returns the contexts if found
 // and a bool indicating success.
 func headContextsNoCost(pr *PullRequest) ([]Context, bool) {
-	if pr.HeadRef != nil {
+	if pr.HeadRef != nil && len(pr.HeadRef.Target.Commit.Status.Contexts) != 0 {
```

Comment:
Can you point me to an example where this is actually helpful? Using the graphql explorer I am only seeing that this field is empty if there actually aren't any status contexts.

Reply:
[example](https://developer.github.com/v4/explorer/?variables=%7B%22query%22%3A%20%22is%3Apr%20repo%3Acjwagner%2Ftestrepo%22%7D&query=query%20%28%24query%3A%20String%21%29%20%7B%0D%0A%20%20search%28type%3A%20ISSUE%2C%20first%3A%20100%2C%20query%3A%20%24query%29%20%7B%0D%0A%20%20%20%20nodes%20%7B%0D%0A%20%20%20%20%20%20...%20on%20PullRequest%20%7B%0D%0A%20%20%20%20%20%20%20%20title%0D%0A%20%20%20%20%20%20%20%20number%0D%0A%20%20%20%20%20%20%20%20headRefOid%0D%0A%20%20%20%20%20%20%20%20headRef%20%7B%0D%0A%20%20%20%20%20%20%20%20%20%20target%20%7B%0D%0A%20%20%20%20%20%20%20%20%20%20%20%20...%20on%20Commit%20%7B%0D%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20oid%0D%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20status%20%7B%0D%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20contexts%20%7B%0D%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20context%0D%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20state%0D%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%7D%0D%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%7D%0D%0A%20%20%20%20%20%20%20%20%20%20%20%20%7D%0D%0A%20%20%20%20%20%20%20%20%20%20%7D%0D%0A%20%20%20%20%20%20%20%20%7D%0D%0A%20%20%20%20%20%20%20%20commits%28last%3A%202%29%20%7B%0D%0A%20%20%20%20%20%20%20%20%20%20nodes%20%7B%0D%0A%20%20%20%20%20%20%20%20%20%20%20%20commit%20%7B%0D%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20oid%0D%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20status%20%7B%0D%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20contexts%20%7B%0D%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20context%0D%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20state%0D%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%7D%0D%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%7D%0D%0A%20%20%20%20%20%20%20%20%20%20%20%20%7D%0D%0A%20%20%20%20%20%20%20%20%20%20%7D%0D%0A%20%20%20%20%20%20%20%20%7D%0D%0A%20%20%20%20%20%20%7D%0D%0A%20%20%20%20%7D%0D%0A%20%20%7D%0D%0A%7D%0D%0A)

Reply:
Above log was with the logging in place but with no `len` checks. Here is one with a  `len` check only on `HeadRef`:

```
$ bazel-bin/prow/cmd/tide/linux_amd64_pure_stripped/tide --dry-run=true --github-token-file=/tmp/token --config-path=/tmp/config.yaml --deck-url=http://prow.build-infra.jetstack.net/ --run-once=true 2>&1 | grep -v CreateStatus | tee /tmp/tide-log.cert.txt
{"client":"github","component":"tide","level":"info","msg":"Throttle(800, 20)","time":"2018-07-11T08:48:33-07:00"}
{"client":"github","component":"tide","level":"info","msg":"Throttle(400, 200)","time":"2018-07-11T08:48:33-07:00"}
{"component":"tide","controller":"sync","level":"debug","msg":"Building tide pool.","time":"2018-07-11T08:48:35-07:00"}
{"component":"tide","controller":"sync","level":"debug","msg":"Search for query \"is:pr state:open repo:\"jetstack/cert-manager\" label:\"lgtm\" label:\"approved\" -label:\"needs-ok-to-test\" -label:\"do-not-merge\" -label:\"do-not-merge/blocked-paths\" -label:\"do-not-merge/cherry-pick-not-approved\" -label:\"do-not-merge/hold\" -label:\"do-not-merge/release-note-label-needed\" -label:\"do-not-merge/work-in-progress\" -label:\"needs-rebase\"\" returned 2 PRs and cost 2 point(s). 4796 remaining.","time":"2018-07-11T08:48:36-07:00"}
{"client":"github","component":"tide","level":"info","msg":"GetRef(jetstack, cert-manager, heads/master)","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","controller":"status-update","level":"debug","msg":"Search for query \"is:pr state:open repo:\"jetstack/cert-manager\"\" returned 35 PRs and cost 2 point(s). 4794 remaining.","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:38-07:00"}
{"component":"tide","controller":"sync","level":"debug","msg":"Firing up 1 goroutines","time":"2018-07-11T08:48:38-07:00"}
{"base-sha":"bd7f15d5f457074c5fd92a60fcc31c8005785531","branch":"master","component":"tide","controller":"sync","level":"info","msg":"Syncing subpool: 2 PRs, 0 PJs.","org":"jetstack","repo":"cert-manager","time":"2018-07-11T08:48:38-07:00"}
{"client":"github","component":"tide","level":"info","msg":"GetPullRequestChanges(jetstack, cert-manager, 612)","time":"2018-07-11T08:48:38-07:00"}
{"client":"github","component":"tide","level":"info","msg":"GetPullRequestChanges(jetstack, cert-manager, 622)","time":"2018-07-11T08:48:38-07:00"}
{"base-sha":"bd7f15d5f457074c5fd92a60fcc31c8005785531","branch":"master","component":"tide","controller":"sync","level":"debug","msg":"missing presubmit cert-manager-chart-verify","org":"jetstack","pr":612,"repo":"cert-manager","sha":"51313b85809341fba8866acdbecc6c2595151b15","time":"2018-07-11T08:48:39-07:00"}
{"base-sha":"bd7f15d5f457074c5fd92a60fcc31c8005785531","branch":"master","component":"tide","controller":"sync","level":"debug","msg":"missing presubmit cert-manager-chart-verify","org":"jetstack","pr":622,"repo":"cert-manager","sha":"bbb65baa3832a1ccf507b4774fc600ae39bbe3b0","time":"2018-07-11T08:48:39-07:00"}
{"base-sha":"bd7f15d5f457074c5fd92a60fcc31c8005785531","branch":"master","component":"tide","controller":"sync","level":"debug","msg":"accumulating PRs for batch testing","org":"jetstack","repo":"cert-manager","time":"2018-07-11T08:48:39-07:00"}
{"base-sha":"bd7f15d5f457074c5fd92a60fcc31c8005785531","batch-passing":null,"batch-pending":null,"branch":"master","component":"tide","controller":"sync","level":"info","msg":"Subpool accumulated.","org":"jetstack","prs-missing":[612,622],"prs-passing":null,"prs-pending":null,"repo":"cert-manager","time":"2018-07-11T08:48:39-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:39-07:00"}
{"base-sha":"bd7f15d5f457074c5fd92a60fcc31c8005785531","branch":"master","component":"tide","controller":"sync","level":"debug","msg":"from 5 total contexts ([cert-manager-e2e-v1-7 cert-manager-e2e-v1-8 cert-manager-e2e-v1-9 cert-manager-quick-verify tide]) found 1 failing contexts: [cert-manager-chart-verify]","org":"jetstack","pr":612,"repo":"cert-manager","sha":"51313b85809341fba8866acdbecc6c2595151b15","time":"2018-07-11T08:48:39-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:39-07:00"}
{"base-sha":"bd7f15d5f457074c5fd92a60fcc31c8005785531","branch":"master","component":"tide","controller":"sync","level":"debug","msg":"from 5 total contexts ([cert-manager-e2e-v1-7 cert-manager-e2e-v1-8 cert-manager-e2e-v1-9 cert-manager-quick-verify tide]) found 1 failing contexts: [cert-manager-chart-verify]","org":"jetstack","pr":622,"repo":"cert-manager","sha":"bbb65baa3832a1ccf507b4774fc600ae39bbe3b0","time":"2018-07-11T08:48:39-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:39-07:00"}
{"base-sha":"bd7f15d5f457074c5fd92a60fcc31c8005785531","branch":"master","component":"tide","controller":"sync","level":"debug","msg":"from 5 total contexts ([cert-manager-e2e-v1-7 cert-manager-e2e-v1-8 cert-manager-e2e-v1-9 cert-manager-quick-verify tide]) found 1 failing contexts: [cert-manager-chart-verify]","org":"jetstack","pr":612,"repo":"cert-manager","sha":"51313b85809341fba8866acdbecc6c2595151b15","time":"2018-07-11T08:48:39-07:00"}
{"component":"tide","level":"debug","msg":"using commit node to return []","time":"2018-07-11T08:48:39-07:00"}
{"base-sha":"bd7f15d5f457074c5fd92a60fcc31c8005785531","branch":"master","component":"tide","controller":"sync","level":"debug","msg":"from 5 total contexts ([cert-manager-e2e-v1-7 cert-manager-e2e-v1-8 cert-manager-e2e-v1-9 cert-manager-quick-verify tide]) found 1 failing contexts: [cert-manager-chart-verify]","org":"jetstack","pr":622,"repo":"cert-manager","sha":"bbb65baa3832a1ccf507b4774fc600ae39bbe3b0","time":"2018-07-11T08:48:39-07:00"}
{"base-sha":"bd7f15d5f457074c5fd92a60fcc31c8005785531","branch":"master","component":"tide","controller":"sync","level":"debug","msg":"of 2 possible PRs, none were passing tests, no batch will be created","org":"jetstack","repo":"cert-manager","time":"2018-07-11T08:48:39-07:00"}
{"action":"WAIT","base-sha":"bd7f15d5f457074c5fd92a60fcc31c8005785531","branch":"master","component":"tide","controller":"sync","level":"debug","msg":"nothing to do","org":"jetstack","repo":"cert-manager","time":"2018-07-11T08:48:39-07:00"}
{"action":"WAIT","base-sha":"bd7f15d5f457074c5fd92a60fcc31c8005785531","branch":"master","component":"tide","controller":"sync","level":"info","msg":"Subpool synced.","org":"jetstack","repo":"cert-manager","targets":null,"time":"2018-07-11T08:48:39-07:00"}
{"component":"tide","duration":"4.497991956s","level":"info","msg":"Synced","time":"2018-07-11T08:48:39-07:00"}
```

Here is a log with both `len` checks:

```
$ bazel-bin/prow/cmd/tide/linux_amd64_pure_stripped/tide --dry-run=true --github-token-file=/tmp/token --config-path=/tmp/config.yaml --deck-url=http://prow.build-infra.jetstack.net/ --run-once=true 2>&1 | grep -v CreateStatus | tee /tmp/tide-log.cert.txt
{"client":"github","component":"tide","level":"info","msg":"Throttle(800, 20)","time":"2018-07-11T08:49:10-07:00"}
{"client":"github","component":"tide","level":"info","msg":"Throttle(400, 200)","time":"2018-07-11T08:49:10-07:00"}
{"component":"tide","controller":"sync","level":"debug","msg":"Building tide pool.","time":"2018-07-11T08:49:11-07:00"}
{"component":"tide","controller":"sync","level":"debug","msg":"Search for query \"is:pr state:open repo:\"jetstack/cert-manager\" label:\"lgtm\" label:\"approved\" -label:\"needs-ok-to-test\" -label:\"do-not-merge\" -label:\"do-not-merge/blocked-paths\" -label:\"do-not-merge/cherry-pick-not-approved\" -label:\"do-not-merge/hold\" -label:\"do-not-merge/release-note-label-needed\" -label:\"do-not-merge/work-in-progress\" -label:\"needs-rebase\"\" returned 2 PRs and cost 2 point(s). 4792 remaining.","time":"2018-07-11T08:49:12-07:00"}
{"client":"github","component":"tide","level":"info","msg":"GetRef(jetstack, cert-manager, heads/master)","time":"2018-07-11T08:49:14-07:00"}
{"component":"tide","controller":"status-update","level":"debug","msg":"Search for query \"is:pr state:open repo:\"jetstack/cert-manager\"\" returned 35 PRs and cost 2 point(s). 4788 remaining.","time":"2018-07-11T08:49:14-07:00"}
{"component":"tide","controller":"sync","level":"debug","msg":"Firing up 1 goroutines","time":"2018-07-11T08:49:14-07:00"}
{"base-sha":"bd7f15d5f457074c5fd92a60fcc31c8005785531","branch":"master","component":"tide","controller":"sync","level":"info","msg":"Syncing subpool: 2 PRs, 0 PJs.","org":"jetstack","repo":"cert-manager","time":"2018-07-11T08:49:14-07:00"}
{"client":"github","component":"tide","level":"info","msg":"GetPullRequestChanges(jetstack, cert-manager, 622)","time":"2018-07-11T08:49:14-07:00"}
{"client":"github","component":"tide","level":"info","msg":"GetPullRequestChanges(jetstack, cert-manager, 612)","time":"2018-07-11T08:49:14-07:00"}
{"base-sha":"bd7f15d5f457074c5fd92a60fcc31c8005785531","branch":"master","component":"tide","controller":"sync","level":"debug","msg":"missing presubmit cert-manager-chart-verify","org":"jetstack","pr":622,"repo":"cert-manager","sha":"bbb65baa3832a1ccf507b4774fc600ae39bbe3b0","time":"2018-07-11T08:49:15-07:00"}
{"base-sha":"bd7f15d5f457074c5fd92a60fcc31c8005785531","branch":"master","component":"tide","controller":"sync","level":"debug","msg":"missing presubmit cert-manager-chart-verify","org":"jetstack","pr":612,"repo":"cert-manager","sha":"51313b85809341fba8866acdbecc6c2595151b15","time":"2018-07-11T08:49:15-07:00"}
{"base-sha":"bd7f15d5f457074c5fd92a60fcc31c8005785531","branch":"master","component":"tide","controller":"sync","level":"debug","msg":"accumulating PRs for batch testing","org":"jetstack","repo":"cert-manager","time":"2018-07-11T08:49:15-07:00"}
{"base-sha":"bd7f15d5f457074c5fd92a60fcc31c8005785531","batch-passing":null,"batch-pending":null,"branch":"master","component":"tide","controller":"sync","level":"info","msg":"Subpool accumulated.","org":"jetstack","prs-missing":[622,612],"prs-passing":null,"prs-pending":null,"repo":"cert-manager","time":"2018-07-11T08:49:15-07:00"}
{"base-sha":"bd7f15d5f457074c5fd92a60fcc31c8005785531","branch":"master","component":"tide","controller":"sync","level":"debug","msg":"from 5 total contexts ([cert-manager-e2e-v1-7 cert-manager-e2e-v1-8 cert-manager-e2e-v1-9 cert-manager-quick-verify tide]) found 1 failing contexts: [cert-manager-chart-verify]","org":"jetstack","pr":622,"repo":"cert-manager","sha":"bbb65baa3832a1ccf507b4774fc600ae39bbe3b0","time":"2018-07-11T08:49:15-07:00"}
{"base-sha":"bd7f15d5f457074c5fd92a60fcc31c8005785531","branch":"master","component":"tide","controller":"sync","level":"debug","msg":"from 5 total contexts ([cert-manager-e2e-v1-7 cert-manager-e2e-v1-8 cert-manager-e2e-v1-9 cert-manager-quick-verify tide]) found 1 failing contexts: [cert-manager-chart-verify]","org":"jetstack","pr":612,"repo":"cert-manager","sha":"51313b85809341fba8866acdbecc6c2595151b15","time":"2018-07-11T08:49:15-07:00"}
{"base-sha":"bd7f15d5f457074c5fd92a60fcc31c8005785531","branch":"master","component":"tide","controller":"sync","level":"debug","msg":"from 5 total contexts ([cert-manager-e2e-v1-7 cert-manager-e2e-v1-8 cert-manager-e2e-v1-9 cert-manager-quick-verify tide]) found 1 failing contexts: [cert-manager-chart-verify]","org":"jetstack","pr":612,"repo":"cert-manager","sha":"51313b85809341fba8866acdbecc6c2595151b15","time":"2018-07-11T08:49:15-07:00"}
{"base-sha":"bd7f15d5f457074c5fd92a60fcc31c8005785531","branch":"master","component":"tide","controller":"sync","level":"debug","msg":"from 5 total contexts ([cert-manager-e2e-v1-7 cert-manager-e2e-v1-8 cert-manager-e2e-v1-9 cert-manager-quick-verify tide]) found 1 failing contexts: [cert-manager-chart-verify]","org":"jetstack","pr":622,"repo":"cert-manager","sha":"bbb65baa3832a1ccf507b4774fc600ae39bbe3b0","time":"2018-07-11T08:49:15-07:00"}
{"base-sha":"bd7f15d5f457074c5fd92a60fcc31c8005785531","branch":"master","component":"tide","controller":"sync","level":"debug","msg":"of 2 possible PRs, none were passing tests, no batch will be created","org":"jetstack","repo":"cert-manager","time":"2018-07-11T08:49:15-07:00"}
{"action":"WAIT","base-sha":"bd7f15d5f457074c5fd92a60fcc31c8005785531","branch":"master","component":"tide","controller":"sync","level":"debug","msg":"nothing to do","org":"jetstack","repo":"cert-manager","time":"2018-07-11T08:49:15-07:00"}
{"action":"WAIT","base-sha":"bd7f15d5f457074c5fd92a60fcc31c8005785531","branch":"master","component":"tide","controller":"sync","level":"info","msg":"Subpool synced.","org":"jetstack","repo":"cert-manager","targets":null,"time":"2018-07-11T08:49:15-07:00"}
{"component":"tide","duration":"3.377686179s","level":"info","msg":"Synced","time":"2018-07-11T08:49:15-07:00"}
```

Reply:
So I guess we saw the length of those node statuses be 0 but we found some other commit node that matched? Since we didn't go to graphql for anything...

Reply:
So in the last case were we requesting the status contexts from the v3 API for all the commits because the headRef and last commit were empty?(!) That should be possible for newly created PRs that haven't had statuses marked as "Job triggered", but that should be a rare case, not the common case.
If so it would be great if you could provide a query that reproduces this in the GitHub graphql explorer to determine if this is Tide problem or a v4 API problem.